src/pages/dev/general-message-passing/solidity-utilities.mdx

```diff
@@ -1,24 +1,74 @@
+
+</Callout>
+
+## Create3 Deployer
+
+Creating an interchain dApp will often require the same contract to be deployed on multiple chains.
 Furthermore, it is useful to know each address of this contract on each chain, either to know where to send remote contract calls, or where to trust remote contract calls from -- often both.
 If we can guarantee that the contracts in question will be deployed at the same address on each network, then the above is trivial.
-This can be achieved by deploying each contract from the same address with the same nonce at each network, or by using [`create2`](https://eips.ethereum.org/EIPS/eip-1014).
-For this purpose, we deployed [`ConstAddressDeployer`](https://github.com/axelarnetwork/axelar-gmp-sdk-solidity/blob/main/contracts/deploy/ConstAddressDeployer.sol) at
-`0x98b2920d53612483f91f12ed7754e51b4a77919e` on every EVM testnet and mainnet that is supported by Axelar.
+This can be achieved by deploying each contract from the same address with the same nonce at each network, or by using [`create3`](https://github.com/axelarnetwork/axelar-gmp-sdk-solidity/blob/main/contracts/deploy/Create3.sol).
+For this purpose, we launched [`Create3 Deployer`](https://github.com/axelarnetwork/axelar-gmp-sdk-solidity/blob/main/contracts/deploy/Create3Deployer.sol) at
+`0xf49B10ccFB7D82C3a8749fFB1aAF3e0c936Eba36` on every EVM testnet and mainnet that is supported by Axelar.
 We plan on deploying it on future supported testnets and mainnets, too.
 
-`ConstAddressDeployer` exposes the following functions:
+`Create3 Deployer` exposes the following functions:
+
+- `deployedAddress(bytes bytecode, bytes32 deploySalt)`: calculates the address of contracts that has been/will be deployed using the contract's bytecode and salt, by a certain sender.
+- `deploy(bytes bytecode, bytes32 deploySalt)`: deploys a contract with a certain bytecode and salt.
+
+#### Constructor Arguments
+
+A key benefit of using `create3` over [`create2`](https://github.com/axelarnetwork/axelar-gmp-sdk-solidity/blob/main/contracts/deploy/Create2.sol) is that you get a constant address independent of its bytecode.
+
+In order to pass in data for a contract's constructor arguments you must encode the values along with the bytecode of the contract.
+
+For example to deploy a contract with a constructor that takes in two arguments, a `uint256` and an `address`, you could call the `deploy()` function as follows:
+
+```
+function deployContract() external {
+    Create3Deployer.deploy(
+        keccak256(bytes("<my salt>")),
+        abi.encodePacked(
+            type(NewContract).creationCode,
+            abi.encode(123, msg.sender)
+        )
+    );
+}
+```
+
+You have the option to either interact with the Solidity functionality directly, but we also provide some scripts. Simply import [`create3Deployer.js`](https://github.com/axelarnetwork/axelar-gmp-sdk-solidity/blob/main/scripts/create3Deployer.js). The following functionality is available:
```

Comment:
some scripts or a script?

"The following functionality is available" is weirdly ambiguous. It sounds like we're trying to show a list of functions available in our library?

Perhaps we should say `the following methods are exported from ...create3Deployer.js...:`